applications/nrf_desktop/src/modules/Kconfig.dvfs

```diff
@@ -41,24 +41,36 @@ config DESKTOP_DVFS_RETRY_COUNT
 	  Number of retries of DVFS frequency change after which DVFS module will report
 	  MODULE_STATE_ERROR.
 
-config DESKTOP_DVFS_SMP_TRANSFER_TIMEOUT_MS
-	int "SMP transfer timeout"
-	default 5000
-	range 1 65535
-	help
-	  Timeout in milliseconds specifying time after which DVFS module will go back to
-	  low frequency after receiving SMP transfer event.
-
-config DESKTOP_DVFS_CONFIG_CHANNEL_TIMEOUT_MS
-	int "Config channel timeout"
-	default 5000
-	range 1 65535
-	help
-	  Timeout in milliseconds specifying time after which DVFS module will go back to
-	  low frequency after reciving config event.
-
 module = DESKTOP_DVFS
 module-str = DVFS
 source "subsys/logging/Kconfig.template.log_config"
 
+# Cannot leave this empty, as it will cause Kconfiglib to fail so add always met dependency
+dvfs_dependencies = DESKTOP_DVFS
+dvfs_state = DVFS_STATE_INITIALIZING
+# We start with DVFS_STATE_INITIALIZING on as it is active on start.
+config DVFS_STATE_INITIALIZING_INITIAL_ACTIVE
+	default y
+rsource "Kconfig.dvfs_state_template"
+
+
+dvfs_dependencies = CAF_BLE_COMMON_EVENTS
+dvfs_state = DVFS_STATE_LLPM_CONNECTED
+choice  DVFS_STATE_LLPM_CONNECTED_ACTIVE_FREQ
+	default DVFS_STATE_LLPM_CONNECTED_FREQ_MEDLOW
+endchoice
+rsource "Kconfig.dvfs_state_template"
+
+dvfs_dependencies = DESKTOP_USB_ENABLE
+dvfs_state = DVFS_STATE_USB_CONNECTED
```

Comment:
Consider hiding the prefix `DVFS_STATE_` inside the Kconfig template (
`menuconfig DVFS_STATE_$(dvfs_state)_ENABLE` etc.) and inside `FOR_DVFS_STATE_LIST` and friends (ex. `MACRO(INITIALIZING) etc.`) to not duplicate it and ensure that it will be always present in names :D

Reply:
These Kconfigs should be prefixed in the same way as in the `Kconfig.dvfs` file with the `DESKTOP_DVFS` prefix.

```
dvfs_state = LLPM_CONNECTED
DESKTOP_DVFS_STATE_$(dvfs_state)_ENABLE
```

Reply:
I have problems with doing it as:

case DVFS_STATE_ ## name: return #DVFS_STATE_ ## #name;)) dosen't work

Reply:
try:
```
#define DEFINE_ENUM(name, ...) _CONCAT(PREFIX_, name),

#define GET_ENUM_NAME(name, ...)				    \
		case _CONCAT(PREFIX_, name): return STRINGIFY(_CONCAT(PREFIX_, name));
```